runtime/criusupport/criusupport.cpp

```diff
@@ -244,11 +248,31 @@ getNativeString(J9VMThread *currentThread, j9object_t javaString, char **nativeS
 	return res;
 }
 
+/**
+ * Checks if a thread is used by JDWP for debugging. Use string compare to check that the thread
+ * name starts with "JDWP" since all JDWP threads have that as a prefix.
+ *
+ * @param[in] currentThread the thread being checked
+ *
+ * @return true if the thread is used by JDWP, false if not
+ */
+static bool
+isJdwpDebugThread(J9VMThread *currentThread)
```

Comment:
This can be expanded to take an additional parameter `toggleDebugThreads` such that it contains the logic of `((!toggleDebugThreads && !isJdwpDebugThread(walkThread)) || (toggleDebugThreads && isJdwpDebugThread(walkThread)))`, and simplify the `toggleThread` check within following `toggleSuspendOnJavaThreads`.